indi-svbony/svbony_base.cpp

```diff
@@ -0,0 +1,1341 @@
+/*
+    SVB Camera Base
+
+    Copyright (C) 2015 Jasem Mutlaq (mutlaqja@ikarustech.com)
+    Copyright (C) 2018 Leonard Bottleman (leonard@whiteweasel.net)
+    Copyright (C) 2021 Pawel Soja (kernel32.pl@gmail.com)
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2.1 of the License, or (at your option) any later version.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library; if not, write to the Free Software
+    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+#include "svbony_base.h"
+#include "svbony_helpers.h"
+
+#include "config.h"
+
+#include <stream/streammanager.h>
+#include <indielapsedtimer.h>
+
+#include <algorithm>
+#include <cmath>
+#include <vector>
+#include <map>
+#include <unistd.h>
+
+#define MAX_EXP_RETRIES         3
+#define VERBOSE_EXPOSURE        3
+#define TEMP_TIMER_MS           1000 /* Temperature polling time (ms) */
+#define TEMP_THRESHOLD          .25  /* Differential temperature threshold (C)*/
+
+#define CONTROL_TAB "Controls"
+
+static bool warn_roi_height = true;
+static bool warn_roi_width = true;
+
+const char *SVBONYBase::getBayerString() const
```

Comment:
If SVBONYBase::getBayerString() is called before the camera is opened, the Bayer Pattern should be "Unknown".
SVBONY Camera SDK cannot get the Bayer Pattern before the camera is opened.